en/docs/learn/password-patterns.md

```diff
@@ -62,6 +62,663 @@ multi-tenant environment.
+                        return super.doPostUpdateRoleListOfUser(userName, deletedRoles, newRoles, userStoreManager);
+                    }
+                
+                    @Override
+                    public boolean doPreGetUserClaimValue(String userName, String claim, String profileName, UserStoreManager storeManager) throws UserStoreException {
+                        return super.doPreGetUserClaimValue(userName, claim, profileName, storeManager);
+                    }
+                
+                    @Override
+                    public boolean doPreGetUserClaimValues(String userName, String[] claims, String profileName, Map<String, String> claimMap, UserStoreManager storeManager) throws UserStoreException {
+                        return super.doPreGetUserClaimValues(userName, claims, profileName, claimMap, storeManager);
+                    }
+                
+                    @Override
+                    public boolean doPostGetUserClaimValue(String userName, String claim, List<String> claimValue, String profileName, UserStoreManager storeManager) throws UserStoreException {
+                        return super.doPostGetUserClaimValue(userName, claim, claimValue, profileName, storeManager);
+                    }
+                
+                    @Override
+                    public boolean doPostGetUserClaimValues(String userName, String[] claims, String profileName, Map<String, String> claimMap, UserStoreManager storeManager) throws UserStoreException {
+                        return super.doPostGetUserClaimValues(userName, claims, profileName, claimMap, storeManager);
+                    }
+                
+                    /**
+                    * Get the logged in user's username who is calling the operation
+                    * @return username
+                    */
+                
+                    private String getUser() {
+                        return CarbonContext.getThreadLocalCarbonContext().getUsername() + "@" +
+                                CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+                    }
+                
+                    private String generatePassword()
+                    {
+                        //implement logic to generate password align with regex pattern
+                        return RandomStringUtils.randomNumeric(12);
+                    }
+            
+                }
+            ```
+    b.  Implement this logic within the `generatePassword()` method.   
+    c.  Build the OSGI bundle of the custom handler and place the .jar file in the `<IS_HOME>/repository/components/dropins` directory.
+    d.  Restart WSO2 Identity Server.
+
+    To check whether the generated password is a randomly generated password, use the http://wso2.org/claims/identity/askPassword claim. 
```

Comment:
Deleted this line since we have already mentioned "temporary passwords are automatically generated" in the beginning of the warning with links to ask password and jit provisioning docs.